arbnode/espresso_snapshot_handler.go

```diff
@@ -0,0 +1,95 @@
+package arbnode
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/offchainlabs/nitro/arbutil"
+	"github.com/offchainlabs/nitro/espresso/authdb"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+)
+
+type EspressoSnapshotHandler struct {
+	stopwaiter.StopWaiter
+	db               *authdb.AuthDB
+	parentChainDir   string
+	l2chainDataDir   string
+	useSnapshot      bool
+	generateSnapshot bool
+}
+
+func NewEspressoSnapshotHandler(db *authdb.AuthDB, parentChainDir string, l2chainDataDir string, useSnapshot bool, generateSnapshot bool) *EspressoSnapshotHandler {
+	return &EspressoSnapshotHandler{
+		db:               db,
+		parentChainDir:   parentChainDir,
+		l2chainDataDir:   l2chainDataDir,
+		useSnapshot:      useSnapshot,
+		generateSnapshot: generateSnapshot,
+	}
+}
+
+func (s *EspressoSnapshotHandler) StoreSnapshotSha256(sum string) error {
+	path := filepath.Join(s.parentChainDir, "snapshot.txt")
+	file, err := os.Create(path)
+	if err != nil {
+		return fmt.Errorf("failed to create snapshot file: %w", err)
+	}
+	defer file.Close()
+	_, err = file.WriteString(sum)
+	if err != nil {
+		return fmt.Errorf("failed to write snapshot file: %w", err)
+	}
+	log.Info("Stored the snapshot hash in a file", "sha256_h1", sum, "path", path)
+	return nil
+}
+
+func (s *EspressoSnapshotHandler) Start(ctx context.Context) error {
+	s.StopWaiter.Start(ctx, s)
+	if !s.useSnapshot {
+		return nil
+	}
+	// Only if snapshot mode is enabled, we re-initialize the tags
+	err := s.db.InitAuthTagsDatabase()
```

Comment:
this is confusing. in the majority of cases, we are not using snapshot? when we just resume from an already tagged db, in those cases `useSnapshot = false` but why are we retagging it?